packages/table/src/table.tsx

```diff
@@ -432,13 +449,23 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
 
     private columnHeaderCellRenderer = (columnIndex: number) => {
         const props = this.getColumnProps(columnIndex);
+        let loading: boolean;
+        switch (this.hasLoadingOption(props.loadingOptions, ColumnLoadingOption.HEADER)) {
```

Comment:
`this.hasLoadingOptions` returns `undefined` is the first argument is `null` or `undefined`. that the `loading` variable can be `undefined` is important. where `false` means "don't show the loading state", `undefined` means defer to the table loading options